src/main/java/io/kuzzle/sdk/core/KuzzleRoom.java

```diff
@@ -194,19 +278,22 @@ public void onSuccess(JSONObject args) {
           } catch (JSONException e) {
             e.printStackTrace();
           }
+
+          KuzzleRoom.this.kuzzle.addSubscription(KuzzleRoom.this.roomId, KuzzleRoom.this.id, KuzzleRoom.this);
           KuzzleRoom.this.kuzzle.getSocket().on(KuzzleRoom.this.channel, new Emitter.Listener() {
             @Override
             public void call(final Object... args) {
-              callAfterRenew(listener, args[0]);
+              callAfterRenew(args[0]);
             }
           });
+
+          KuzzleRoom.this.dequeue();
         }
 
         @Override
         public void onError(JSONObject arg) {
-          if (listener != null) {
-            listener.onError(arg);
-          }
+          KuzzleRoom.this.queue.clear();
+          listener.onError(arg);
```

Comment:
Should set this.subscribing to false before calling the callback in case of error